### ISSUE TITLE
TypeError when fetching a list on the second time

### ISSUE THREAD

**Issue by miketaylr** [past/tsci#48]
This probably is due to the changes I introduced for #44 #46, etc.

```
Downloaded Tranco list with ID 4QQX for date 2019-05-12
(node:52424) UnhandledPromiseRejectionWarning: TypeError: Expected a string
    at module.exports (/Users/miket/dev/tsci/node_modules/escape-string-regexp/index.js:7:9)
    at /Users/miket/dev/tsci/tranco.js:22:65
    at Array.forEach (<anonymous>)
    at /Users/miket/dev/tsci/tranco.js:18:29
    at new Promise (<anonymous>)
    at removeIgnoredDomains (/Users/miket/dev/tsci/tranco.js:15:12)
    at WriteStream.<anonymous> (/Users/miket/dev/tsci/tranco.js:92:17)
    at WriteStream.emit (events.js:205:15)
    at finishMaybe (_stream_writable.js:644:14)
    at _stream_writable.js:622:5
(node:52424) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:52424) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

If I delete that downloaded CSV and restart it, it runs fine. I've only noticed this when trying to do an month, when it got to `2019-05-12` it died.

**Comment by miketaylr** [comment]
Oh, WAIT. I know the problem. If you do a single run, you won't ever see the bug.

```
IGNORED_DOMAINS.forEach((value, index) => {
                // create an escaped regexp out of each domain we want to ignore
                // the CSV format will look like the following:
                // 1,example.com\r\n
                IGNORED_DOMAINS[index] = new RegExp(`\\d{1,3},${escapeStringRegexp(value)}\\r?\\n`);
            });
```

Basically we're changing `IGNORED_DOMAINS` from an array of strings to an array of regexps, so when it runs again we're asking `escapeStringRegexp` to escape... a regexp. We can just use map here to return a new array that way we're not modifying the original (or just cache the modified array).

**Comment by miketaylr** [comment]
I can fix this in #47.
